include/mbedtls/config_psa.h

```diff
@@ -40,6 +40,58 @@ extern "C" {
 
 #if defined(MBEDTLS_PSA_CRYPTO_CONFIG)
 
+/*
+ * If the configuration has requested using PSA crypto config,
+ * undefine any other definitions that may have defined these
+ * features, so that they are strictly determined by the PSA config.
+ */
+#undef MBEDTLS_PSA_BUILTIN_ALG_DETERMINISTIC_ECDSA
+#undef MBEDTLS_ECDSA_DETERMINISTIC
```

Comment:
Undefining `MBEDTLS_xxx` here is definitely wrong. Consider an application that uses both `mbedtls_pk_sign` and 
`psa_sign_hash` (perhaps via two different libraries). Its `config.h` will need to enable both `MBEDTLS_ECDSA_C` and `PSA_WANT_ALG_ECDSA`. This is true regardless of the configuration mode.